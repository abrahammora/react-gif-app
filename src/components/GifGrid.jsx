import PropTypes from 'prop-types';
import { useFetchGifs } from "../hooks/useFetchGifs";
import { Gifitem } from "./Gifitem";


export const GifGrid = ({category}) => {

    // Custom Hook
    const {images, isLoading} = useFetchGifs(category);
   
    
    return (
        <>
            <h3>{ category }</h3>  
            {
                isLoading && (<h2>Cargando..</h2>)
            }
            
            <div className="card-grid">
                {/* images.map */}
                {
                    images.map( (image) => (
                        <Gifitem key={image.id}
                            {...image}
                        />
                        
                    ))
                }
            </div>                    
        </>
    )
}

GifGrid.protype = {
    catgegory: PropTypes.string.isRequired
}
