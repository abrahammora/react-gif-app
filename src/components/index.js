/**
 *  Archivo de exportaciones globales
 * 
 */
export * from './AddCategory';
export * from './GifGrid';
export * from './Gifitem';