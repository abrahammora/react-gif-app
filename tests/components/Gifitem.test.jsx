import { render, screen } from "@testing-library/react";
import { Gifitem } from "../../src/components/Gifitem";

describe('Pruebas en <Gifitem />', () => { 
    const title = 'Saitama';
    const url = 'https://one-punch.com/saitam.jpg'
    test('should match snapshot', () => {
        const {container} =  render(<Gifitem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('should show image with url and alt', () => {
        render(<Gifitem title={title} url={url}/>);
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });


    test('should be the title in the component', () => { 
        render(<Gifitem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });
});