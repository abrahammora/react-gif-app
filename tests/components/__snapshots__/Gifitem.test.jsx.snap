// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Gifitem /> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitam.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
